src/meta-srv/src/procedure/region_migration.rs

```diff
@@ -86,6 +87,8 @@ pub struct PersistentContext {
     /// The timeout for downgrading leader region and upgrading candidate region operations.
     #[serde(with = "humantime_serde", default = "default_timeout")]
     timeout: Duration,
+    /// The trigger reason of region migration.
+    trigger_reason: RegionMigrationTriggerReason,
```

Comment:
For backward compatibility, new added fields should have default values.
```suggestion
    /// The trigger reason of region migration.
    #[serde(default = "RegionMigrationTriggerReason::unknown_reason")]
    trigger_reason: RegionMigrationTriggerReason,
```

Reply:
I add the new enum value `Unknown` in `RegionMigrationTriggerReason`:

```rust
/// The reason why the region migration procedure is triggered.
#[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, strum::Display)]
#[strum(serialize_all = "PascalCase")]
pub enum RegionMigrationTriggerReason {
    #[default]
    /// The region migration procedure is triggered by unknown reason.
    Unknown,
    /// The region migration procedure is triggered by administrator.
    Manual,
    /// The region migration procedure is triggered by auto rebalance.
    AutoRebalance,
    /// The region migration procedure is triggered by failover.
    Failover,
}
```

And mark it `#[serde(default)]` in `PersistentContext`:

```rust
pub struct PersistentContext {
...
    #[serde(default)]
    trigger_reason: RegionMigrationTriggerReason,
}
```